Bio/NMR/xpktools.py

```diff
@@ -179,18 +179,18 @@ def residue_dict(self, index):
     def write_header(self, outfn):
         """Write header lines from input file to handle `outfn`."""
         with open(outfn, 'wb') as outfile:
-            outfile.write(self.firstline)
-            outfile.write("\012")
-            outfile.write(self.axislabels)
-            outfile.write("\012")
-            outfile.write(self.dataset)
-            outfile.write("\012")
-            outfile.write(self.sw)
-            outfile.write("\012")
-            outfile.write(self.sf)
-            outfile.write("\012")
-            outfile.write(self.datalabels)
-            outfile.write("\012")
+            outfile.write(self.firstline.encode())
```

Comment:
This looks wrong - why add ``.encode()`` to all the lines, can't you write the file in text mode?

This file dates back well before Python 3 - most likely using binary mode was an oversight, or is to do with Windows new lines?

Reply:
Because it rises an Type Error; I had to change for it to run, I'm using python 3.6.

Reply:
Does changing ``with open(outfn, 'wb') as outfile:`` to ``with open(outfn, 'w') as outfile:`` work (instead of adding the ``.encode()`)?

Reply:
It doesn't work, it seems as there is no difference of using 'wb'or 'b', if I use open(outfn, 'w') raises TypeError as below:
TypeError: a bytes-like object is required, not 'str'

Reply:
I'm confused, ``self.firstline`` should be string not bytes since the parsing is done in text mode.

As an aside, the use of ``"\012"`` is unusual, ``"\n"`` would be preferable  style wise.

Reply:
The only reason I can think of to explicitly use "\012" or similar is to avoid getting CR LF on Windows - does the command line tool work on Windows? Can you test that? Perhaps leaving it as "\012" would be safer after all.

Reply:
I'll test on Windows, is going to take few days.

Reply:
I manage to remove the "encode" and "b", I think that was my mistake, as I have been running from install while testing DocTest, I've changed the test to check the content instead, as the file size reports different, from the reference created in linux with the file created on Windows, I think  due to CR LF, as content is the same. Now is working on Linux Ubuntu 18.04, python 3.6 and Windows 7 Python 3.7.
Could you please check the merge; the updated commits appear now on my branch, I do not remember seen this on previous merges.